hazelcast-client/src/main/java/com/hazelcast/client/cache/impl/NearCachedClientCacheProxy.java

```diff
@@ -588,7 +588,7 @@ private void removeInvalidationListener() {
 
         String registrationId = nearCacheMembershipRegistrationId;
         if (registrationId != null) {
-            getContext().getRepairingTask(getServiceName()).deregisterHandler(name);
+            getContext().getRepairingTask(getServiceName()).deregisterHandler(nameWithPrefix);
```

Comment:
The handler is registered with `nameWithPrefix`, so this didn't remove the handler.